[DBG_X] Undefined variable ekin_ in TrackingAction.cc

The build of CMSSW_12_5_DBG_X_2022-06-02-2300 IB failed with the following error:
```
/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc10/external/gcc/10.3.0-84898dea653199466402e67d73657f10/bin/c++ -c -DGNU_GCC -D_GNU_SOURCE -DEIGEN_DONT_PARALLELIZE -DTBB_USE_GLIBCXX_VERSION=100300 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DGNU_GCC -DG4V9 -DGNU_GCC -DG4V9 -DDD4HEP_USE_GEANT4_UNITS=1 -DCMSSW_GIT_HASH='CMSSW_12_5_DBG_X_2022-06-02-2300' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_12_5_DBG_X_2022-06-02-2300' -I/data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/5d81603e13eb4ff6d359b7051c5c6210/opt/cmssw/el8_amd64_gcc10/cms/cmssw/CMSSW_12_5_DBG_X_2022-06-02-2300/src -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc10/external/dd4hep/v01-19x-3ca297e974f1429634d596d2fb748f20/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc10/external/geant4/10.7.2-f2f6f28d53526cbce934633504d0a84a/include/Geant4 -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc10/external/geant4/10.7.2-f2f6f28d53526cbce934633504d0a84a/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc10/external/pcre/8.43-28d54724578a9006cf0dca75629374e3/include -isystem/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc10/external/boost/1.78.0-7b33f741d272fd812134b393a416f203/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc10/external/bz2lib/1.0.6-86270889250d9cb8193d1963706a39cb/include -isystem/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc10/external/clhep/2.4.5.1-5bba156cb451d19d7be7ff651e7c5f9f/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc10/external/expat/2.1.0-e2e74851cebffc0aa0988d120ae74fcd/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc10/external/gsl/2.6-84974e6daa70611c23ebff30cbb797c4/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc10/external/hepmc/2.06.10-46867a6dcc6e5712b7953fe57085fcbd/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc10/external/heppdt/3.04.01-c965e2de7fbf97bcecf457e1ea298571/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc10/external/libuuid/2.34-fe91b21b8e54a48f2d45894d9da65f14/include -isystem/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc10/lcg/root/6.24.07-4d9e0be0954b324b55d0784fdb280b83/include -isystem/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc10/external/tbb/v2021.5.0-e966a5acb1e4d5fd7605074bafbb079c/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc10/external/vecgeom/v1.1.17-e658aebf667adad0ef50a21a40815b01/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc10/external/vecgeom/v1.1.17-e658aebf667adad0ef50a21a40815b01/include/VecGeom -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc10/external/xerces-c/3.1.3-5b1e4ab8f5a2057565f3be0ae6a10b37/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc10/external/xz/5.2.5-d6fed2038c4e8d6e04531d1adba59f37/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc10/external/zlib/1.2.11-71514f01d2850dfd0bcd391557259a58/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc10/external/eigen/82dd3710dac619448f50331c1d6a35da673f764a-01f0fbbc69fe5c11dd0acfe3c748f30b/include/eigen3 -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc10/external/fmt/8.0.1-a90b32563595b499263c17a8753d8f1d/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc10/external/hepmc3/3.2.5-c3cd50aeecf06b194814f1a75bf7872e/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc10/external/md5/1.0.0-0440f43b57c5656322f7cd7a65f85771/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc10/external/OpenBLAS/0.3.15-a8c37ec25750b8edfd2c93d1f2e8e582/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc10/external/tinyxml2/6.2.0-b4ef6e4242045ed0c243ef457f5b32a5/include -O2 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++1z -ftree-vectorize -Wstrict-overflow -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -msse3 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-deprecated-copy -Wno-unused-parameter -Wunused -Wparentheses -Wno-deprecated -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -Wno-error=unused-variable -ftls-model=global-dynamic -pthread -DG4GEOM_USE_USOLIDS -DBOOST_DISABLE_ASSERTS -g -O3 -DEDM_ML_DEBUG -fno-lto -Wl,--exclude-libs,ALL  -fPIC   /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/5d81603e13eb4ff6d359b7051c5c6210/opt/cmssw/el8_amd64_gcc10/cms/cmssw/CMSSW_12_5_DBG_X_2022-06-02-2300/src/SimG4Core/Application/src/TrackingAction.cc -o tmp/el8_amd64_gcc10/src/SimG4Core/Application/src/SimG4CoreApplication/bigobj/TrackingAction.cc.o
/data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/5d81603e13eb4ff6d359b7051c5c6210/opt/cmssw/el8_amd64_gcc10/cms/cmssw/CMSSW_12_5_DBG_X_2022-06-02-2300/src/SimG4Core/Application/src/TrackingAction.cc: In member function 'virtual void TrackingAction::PostUserTrackingAction(const G4Track*)':
  /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/5d81603e13eb4ff6d359b7051c5c6210/opt/cmssw/el8_amd64_gcc10/cms/cmssw/CMSSW_12_5_DBG_X_2022-06-02-2300/src/SimG4Core/Application/src/TrackingAction.cc:127:95: error: 'ekin_' was not declared in this scope
   127 |     edm::LogVerbatim("SimTrackManager") << "TrackingAction addTrack " << id << " E(GeV)= " << ekin_ << "  "
      |                                                                                               ^~~~~
  gmake: *** [tmp/el8_amd64_gcc10/src/SimG4Core/Application/src/SimG4CoreApplication/bigobj/TrackingAction.cc.o] Error 1
```

It seems that the line https://github.com/cms-sw/cmssw/blob/master/SimG4Core/Application/src/TrackingAction.cc#L127 should read either
```cpp
edm::LogVerbatim("SimTrackManager") << "TrackingAction addTrack " << id << " E(GeV)= " << ekin << "  "
```
(`ekin` without underscore) or
```cpp
edm::LogVerbatim("SimTrackManager") << "TrackingAction addTrack " << id << " E(GeV)= " << ekin / CLHEP::GeV << "  "
```
(`ekin` without underscore and divided by `CLHEP::GeV`).

Pinging @civanch for comments

--- a/SimG4Core/Application/src/TrackingAction.cc
+++ b/SimG4Core/Application/src/TrackingAction.cc
@@ -124,7 +124,7 @@ void TrackingAction::PostUserTrackingAction(const G4Track* aTrack) {
     eventAction_->addTrack(currentTrack_, true, withAncestor);
 
 #ifdef EDM_ML_DEBUG
-    edm::LogVerbatim("SimTrackManager") << "TrackingAction addTrack " << id << " E(GeV)= " << ekin_ << "  "
+    edm::LogVerbatim("SimTrackManager") << "TrackingAction addTrack " << id << "  "
                                         << aTrack->GetDefinition()->GetParticleName() << " added= " << withAncestor
                                         << " at " << aTrack->GetPosition();
 #endif

